src/Interactive/Host/Interactive/Core/InteractiveHost.cs

```diff
@@ -413,25 +369,38 @@ private async Task<InitializedRemoteService> TryGetOrCreateRemoteServiceAsync()
             return default;
         }
 
-        private async Task<TResult> Async<TResult>(Action<Service, RemoteAsyncOperation<TResult>> action)
+        private async Task<TResult> Async<TResult>(string targetName, params object[] arguments)
         {
+            var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
+            TaskCompletionSource<TResult> _completion = new TaskCompletionSource<TResult>();
+            EventHandler _processExitedHandler = new EventHandler((_, __) =>
+            {
+                initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                _completion.TrySetResult(default(TResult));
+            });
+
             try
             {
-                var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
-                if (initializedService.ServiceOpt == null)
+                initializedService.ServiceOpt.Process.Exited += _processExitedHandler;
+                if (!initializedService.ServiceOpt.Process.IsAlive())
                 {
-                    return default;
+                    initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                    _completion.TrySetResult(default(TResult));
                 }
 
-                return await new RemoteAsyncOperation<TResult>(initializedService.ServiceOpt).ExecuteAsync(action).ConfigureAwait(false);
+                return await initializedService.ServiceOpt._jsonRpc.InvokeAsync<TResult>(targetName, arguments).ConfigureAwait(false);
             }
-            catch (Exception e) when (FatalError.Report(e))
+            catch (RemotingException) when (!initializedService.ServiceOpt.Process.IsAlive())
             {
-                throw ExceptionUtilities.Unreachable;
+                // the operation might have terminated the process:
+                initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                _completion.TrySetResult(default(TResult));
+                return await _completion.Task.ConfigureAwait(false);
```

Comment:
This should just be `return default`. Remember - the completion was only used in the remote operation to facilitate async that isn't available in the previously use remoting framework. Since we're using JSON-RPC and it supports async directly we don't need the completion source here. We just away InvokeAsync and it will do the right thing. If InovkeAsync fails (throws an exception) that means the server either disconnected (process terminated) or we have a bug somewhere. If the process is terminated (which we detect by checking IsAlive) then we previously returned default(TResult) via the completion source. We can do so directly here.